dattri/task.py

```diff
@@ -0,0 +1,238 @@
+            Dict[str, torch.Tensor],
+        ],
+    ) -> None:
+        """Initialize the AttributionTask.
+
+        Args:
+            target_func (Callable): The target function to be attributed.
+                The function can be quite flexible in terms of what is calculate,
+                but it should take the parameters and the data as input. Other than
+                that, the forwarding of model should be in `torch.func` style.
+                A typical example is as follows:
+                ```python
+                def f(params, data):
+                    image, label = data
+                    loss = nn.CrossEntropyLoss()
+                    yhat = torch.func.functional_call(model, params, image)
+                    return loss(yhat, label)
+                ```.
+                This examples calculates the CE loss of the model on the data.
+            model (nn.Module): The model that the target function is based on.
+                To be more specific, the model is the `model` used in the target
+                function. Since only the computation graph of the model will be
+                used, so it is allowed that this model is not loaded with a trained
+                parameters.
+            checkpoints:
+                (Union[str, List[str], List[Dict[str, torch.Tensor]],
+                Dict[str, torch.Tensor]]): The checkpoints
+                of the model, both dictionary of the state_dict and the path to
+                the checkpoint are supported. If ensemble is needed, a list of
+                checkpoint is also supported.
+
+        """
+        self.model = model
+        self.target_func = flatten_func(self.model)(target_func)
+
+        if not isinstance(checkpoints, list):
+            self.checkpoints = [checkpoints]
+        else:
+            self.checkpoints = checkpoints
+
+        # current_checkpoint_idx is used to state
+        # which checkpoint is currently loaded.
+        self.current_checkpoint_idx = None
+
+        # TODO: Make this more general, that is allow customized kwargs.
+        self.grad_func = vmap(grad(self.target_func), in_dims=(None, 1))
+
+    def _load_checkpoints(self, index: int) -> None:
+        """This method load the checkpoint at the specified index.
+
+        Args:
+            index (int): The index of the checkpoint to be loaded.
+        """
+        if self.current_checkpoint_idx is None or self.current_checkpoint_idx != index:
+            if isinstance(self.checkpoints[index], str):
+                self.model.load_state_dict(torch.load(self.checkpoints[index]))
+            else:
+                self.model.load_state_dict(self.checkpoints[index])
+            self.current_checkpoint_idx = index
+            self.named_parameters = {
+                k: p for k, p in self.model.named_parameters() if p.requires_grad
+            }
+
+    @staticmethod
+    def _genearte_param_layer_map(
+        named_parameters: Dict[str, torch.Tensor],
```

Comment:
No I think we can't, or we need some other parameters in this function. Because users may just want to carry out TDA methods on a portion of the parameters (e.g., some specific layers), so named_parameters may only contain a portion of layers.